pkg/eval/builtin_fn_str.go

```diff
@@ -49,14 +49,31 @@ func toString(fm *Frame, args ...any) error {
 // greater than 36.
 var ErrBadBase = errors.New("bad base")
 
-func base(fm *Frame, b int, nums ...int) error {
+// ErrInvalidInput is thrown by the "base" builtin if the input num is invalid
+var ErrInvalidInput = errors.New("invalid input")
+
+func base(fm *Frame, b int, nums ...vals.Num) error {
 	if b < 2 || b > 36 {
 		return ErrBadBase
 	}
 
 	out := fm.ValueOutput()
 	for _, num := range nums {
-		err := out.Put(strconv.FormatInt(int64(num), b))
+		bigInt := new(big.Int)
+		switch num := num.(type) {
+		case int:
+			bigInt.SetInt64(int64(num))
+		case *big.Int:
+			bigInt.Set(num)
+		case string:
```

Comment:
This case is impossible. The `vals.Num` constraint will cause the Elvish runtime to convert any string into a `vals.Num` if it is a valid number. It's helpful to run `make cover` to catch such mistakes.

Reply:
Done.